### DISCUSSION TITLE
delete DAP "Terminate" keymap

### DISCUSSION THREAD

**Discussion by hotgazpacho** [LazyVim/LazyVim#647] [Q&A] [answered]
I'm using `lazyvim.plugins.extras.dap.core`, and I'd like to remove the `<leader>dt` keymap, move it to `<leader>dx`, and then add file-type specific keymaps bound to `<leader>dt` as "Debug Test".

So far, I've put this in my `lua/config/keymaps.lua` file:

```lua
-- dap
vim.keymap.del("n", "<leader>dt")
vim.keymap.set("n", "<leader>dx", function()
  require("dap").terminate()
end, { desc = "Terminate" })
```

I would have expected the mapping to be gone, but which-key still shows it:

<img width="1362" alt="image" src="https://user-images.githubusercontent.com/7071/233501678-10a92809-ec7e-4bd1-9d4f-3082aac3bff5.png">

Is there a way to accomplish this?

**Comment by odysseyalive** [comment]
Use the DAP Neovim Lua Adapter as a reference.  See [here](https://github.com/mfussenegger/nvim-dap) and check the full-spec next to nvim-dap. You'll see an example on setting which-key in the same context.  You'll probably have to blank this out from within the which-key system, though -- not vim.

**Comment by dpetka2001** [comment, marked as answer]
This must be happening because when `nvim-dap` is loaded from `extras.dap.core` it also loads the keymap again. See [here](https://www.lazyvim.org/plugins/extras/dap.core#nvim-dap). So you should just change the keymap in the `keys` section in your own spec. Just put a file in `~/.config/nvim/lua/plugins`
```lua
return {
  {
    "mfussenegger/nvim-dap",
    dependencies = {
      -- which key integration
      {
        "folke/which-key.nvim",
        opts = {
          defaults = {
            ["<leader>dt"] = { name = "+Debug Test" },
          },
        },
      },
    },
    keys = {
      { "<leader>dt", false },
      {
        "<leader>dx",
        function()
          require("dap").terminate()
        end,
        desc = "Terminate",
      },
    },
  },
}
```
Test it out to see if it works for your needs.

**Reply by hotgazpacho** [reply]
That was it! Thank you!
